pkg/auth/service.go

```diff
@@ -125,340 +129,401 @@ func fieldByTag(t reflect.Type, key, tag string) string {
-	if err != nil {
-		return nil, err
 	}
-	return model.ConvertUser(user), nil
-}
 
-func getGroup(tx db.Tx, groupDisplayName string) (*model.DBGroup, error) {
-	group := &model.DBGroup{}
-	err := tx.Get(group, `SELECT * FROM auth_groups WHERE display_name = $1`, groupDisplayName)
-	if err != nil {
-		return nil, err
-	}
-	return group, nil
-}
-
-func getPolicy(tx db.Tx, policyDisplayName string) (*model.DBPolicy, error) {
-	policy := &model.DBPolicy{}
-	err := tx.Get(policy, `SELECT * FROM auth_policies WHERE display_name = $1`, policyDisplayName)
-	if err != nil {
-		return nil, err
-	}
-	return policy, nil
-}
-
-func deleteOrNotFound(tx db.Tx, stmt string, args ...interface{}) error {
-	res, err := tx.Exec(stmt, args...)
-	if err != nil {
-		return err
+	var entries []proto.Message
+	p := &model.Paginator{}
+	for len(entries) <= amount && it.Next() {
+		entry := it.Entry()
+		value := entry.Value
+		if len(entries) == amount {
+			p.NextPageToken = entry.Key
+			break
+		}
+		entries = append(entries, value)
 	}
-	numRows := res.RowsAffected()
-	if numRows == 0 {
-		return ErrNotFound
+	if err = it.Err(); err != nil {
+		return nil, nil, fmt.Errorf("list DB: %w", err)
 	}
-	return nil
+	p.Amount = len(entries)
+	return entries, p, nil
 }
 
-type DBAuthService struct {
-	db          db.Database
+type KVAuthService struct {
+	store       kv.StoreMessage
 	secretStore crypt.SecretStore
 	cache       Cache
 	log         logging.Logger
 }
 
-func NewDBAuthService(db db.Database, secretStore crypt.SecretStore, cacheConf params.ServiceCache, logger logging.Logger) *DBAuthService {
+func NewKVAuthService(store kv.StoreMessage, secretStore crypt.SecretStore, cacheConf params.ServiceCache, logger logging.Logger) *KVAuthService {
 	logger.Info("initialized Auth service")
 	var cache Cache
 	if cacheConf.Enabled {
 		cache = NewLRUCache(cacheConf.Size, cacheConf.TTL, cacheConf.EvictionJitter)
 	} else {
 		cache = &DummyCache{}
 	}
-	return &DBAuthService{
-		db:          db,
+	return &KVAuthService{
+		store:       store,
 		secretStore: secretStore,
 		cache:       cache,
 		log:         logger,
 	}
 }
 
-func (s *DBAuthService) decryptSecret(value []byte) (string, error) {
-	decrypted, err := s.secretStore.Decrypt(value)
+func decryptSecret(s crypt.SecretStore, value []byte) (string, error) {
+	decrypted, err := s.Decrypt(value)
 	if err != nil {
 		return "", err
 	}
 	return string(decrypted), nil
 }
 
-func (s *DBAuthService) encryptSecret(secretAccessKey string) ([]byte, error) {
-	encrypted, err := s.secretStore.Encrypt([]byte(secretAccessKey))
+func encryptSecret(s crypt.SecretStore, secretAccessKey string) ([]byte, error) {
+	encrypted, err := s.Encrypt([]byte(secretAccessKey))
 	if err != nil {
 		return nil, err
 	}
 	return encrypted, nil
 }
 
-func (s *DBAuthService) SecretStore() crypt.SecretStore {
-	return s.secretStore
+func (s *KVAuthService) ConvertCredDataList(creds []proto.Message) []*model.Credential {
```

Comment:
In my opinion, this and CredentialFromProto should be in model.go and be package functions